client/src/components/HistoryViewer/HistoryViewer.js

```diff
@@ -29,207 +29,159 @@ import i18n from 'i18n';
  * to be connected providing the query implementation for the appropriate
  * DataObject type
  */
-class HistoryViewer extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleSetPage = this.handleSetPage.bind(this);
-    this.handleAfterRevert = this.handleAfterRevert.bind(this);
-    this.refreshVersionData = this.refreshVersionData.bind(this);
-
-    this.state = {
-      versions: [],
-      totalCount: 0,
-    };
-  }
-
-  componentDidMount() {
-    this.refreshVersionData();
-    // Display a toast if there were any pre-existing graphql errors
-    const { graphQLErrors, toastActions } = this.props;
-    if (graphQLErrors.length > 0) {
-      toastActions.error(i18n._t('Admin.UNKNOWN_ERROR', 'An unknown error has occurred.'));
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    // Display a toast if there were any new graphql errors
-    if (prevProps.graphQLErrors.length < this.props.graphQLErrors.length) {
-      this.props.toastActions.error(i18n._t('Admin.UNKNOWN_ERROR', 'An unknown error has occurred.'));
-    }
-
-    // Manually handle state changes in the page number,
-    if (this.state.versions.length === 0) {
-      return;
-    }
-    if (prevProps.page !== this.props.page) {
-      this.refreshVersionData();
-    }
-  }
-
-  /**
-   * Reset the selected version when unmounting HistoryViewer to prevent data leaking
-   * between instances
-   */
-  componentWillUnmount() {
-    const { onSelect } = this.props;
-    if (typeof onSelect === 'function') {
-      onSelect(0);
-    }
-  }
+const HistoryViewer = ({
+  loading = false,
+  graphQLErrors = [],
+  compare = {},
+  currentVersion = false,
+  isInGridField = false,
+  isPreviewable = false,
+  schemaUrl = '',
+  ListComponent,
+  VersionDetailComponent,
+  CompareWarningComponent,
+  limit,
+  recordId,
+  page,
+  isRevertable,
+  actions,
+  onSelect,
+  onSetPage,
+  onResize,
+  toastActions,
+  recordClass,
+  recordClassSingularName,
+  previewState,
+}) => {
+  const [versions, setVersions] = useState([]);
+  const [totalCount, setTotalCount] = useState(0);
+
+  // Use a ref to track previous error count without causing re-renders
+  const prevErrorCountRef = useRef(0);
```

Comment:
This is new.

I am asking you again to please point out when adding things that weren't there before, including what they're used for and why you needed to add them. It will make reviewing a lot easier if I don't have to figure this stuff out for myself.

Reply:
In this case I think it's not needed - see relevant comment I added for the `useEffect`.

Reply:
Removed